### PR TITLE
resource_control: add quota limiter per priority

### PR DESCRIPTION
**Pull request by glorv**
### What is changed and how it works?

Issue Number: ref #15917

What's Changed:

This PR is the first phase for #15917, it only add the ResourceLimiter and but only enable them as all quotas are set to infinity. I will add auto-tuning the quota in the next PR.

After this change, if `resource control` is enabled and there are at least 1 user defined resource group, all foreground read(including the read part in txn-scheduler) will be apply with an extra cpu quota-limiter. By setting proper quota to the limiter of each priority we can adjust the resource usage of each priority to ensure high-priority tasks alway have enough resources.

```
Add a ResourceLimiter for each priority, the `ResourceLimiter` is used to control the foreground read and txn-read tasks. All tasks with the same priority(explicit in the request or implicit from resource group) share the same Limiter.
```

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test

Side effects

- Performance regression
    - Consumes more CPU

### Release note 

```release-note
None
```

### CODE REVIEW

**Comment by glorv** [comment]
@nolouch  @Connor1996 PTAL

**Comment by glorv** [comment]
@nolouch @Connor1996 PTAL

File: components/resource_control/src/resource_group.rs [hunk 1]
```diff
@@ -56,21 +57,56 @@ pub enum ResourceConsumeType {
...
     // auto incremental version generator used for mark the background
     // resource limiter has changed.
     version_generator: AtomicU64,
+    // the shared resource limiter of each priority
+    priority_limiters: [Arc<ResourceLimiter>; TaskPriority::COUNT],
 }
 
 impl Default for ResourceGroupManager {
     fn default() -> Self {
+        let priority_limiters = ["high", "medium", "low"].map(|n| {
```
Comment (nolouch) [inline components/resource_control/src/resource_group.rs:96]: How about using the defined enum:
```
vec![
    TaskPriority::High,
    TaskPriority::Medium,
    TaskPriority::Low,
]
.into_iter()
.map(|priority| {
    let priority_str = priority.to_str();
    Arc::new(ResourceLimiter::new(
        priority_str.into(),
        f64::INFINITY,
        f64::INFINITY,
        0,
    ))
})
```

**Review by nolouch — APPROVED** [review]
lgtm

**Comment by glorv** [comment]
@Connor1996 PTAL

**Review by Connor1996 — APPROVED** [review]
LGTM

**Pull request merged**